alexk_low_cost_robot/wizard/src/main.rs

```diff
@@ -0,0 +1,314 @@
+            .expect("Communication error");
+    } else {
+        xl430::sync_write_torque_enable(&io, serial_port, &[1, 2], &[0; 2])
+            .expect("Communication error");
+
+        xl430::sync_write_operating_mode(&io, serial_port, &[1, 2], &[4; 2])
+            .expect("Communication error");
+    }
+
+    set_drive_mode(io, serial_port, puppet, &vec![false; 6]);
+    set_homing(io, serial_port, puppet, &vec![0; 6]);
+}
+
+// To register position during the process we need to know approximately in which position the arm is
+fn get_nearest_rounded_positions(pos: &Vec<i32>) -> Vec<i32> {
+    pos.iter()
+        .map(|&x| {
+            if x >= 0 {
+                let k = x / 1024;
+                if x % 1024 > 512 {
+                    (k + 1) * 1024
+                } else {
+                    k * 1024
+                }
+            } else {
+                let k = (-x) / 1024;
+                if (-x) % 1024 > 512 {
+                    -(k + 1) * 1024
+                } else {
+                    -k * 1024
+                }
+            }
+        })
+        .collect::<Vec<_>>()
+}
+
+// useful function to invert the mode of the servos
+fn invert(pos: &Vec<i32>, inverted: &Vec<bool>) -> Vec<i32> {
+    pos.iter()
+        .enumerate()
+        .map(|(i, &x)| if inverted[i] { x } else { -x })
+        .collect::<Vec<_>>()
+}
+
+fn configure_homing(
+    io: &DynamixelSerialIO,
+    serial_port: &mut dyn SerialPort,
+    inverted: &Vec<bool>,
+    puppet: bool,
+) {
+    println!("------Configuring homing");
+
+    // set homing position to 0 for all servos
+    set_homing(io, serial_port, puppet, &vec![0; 6]);
+
+    // get current positions
+    let pos = get_positions(io, serial_port, puppet);
+
+    // get nearest rounded positions
+    let nearest_rounded = get_nearest_rounded_positions(&pos);
+
+    // get correction
+    let correction = get_correction(&nearest_rounded, inverted);
+
+    // set homing position
+    set_homing(io, serial_port, puppet, &correction);
+}
+
+fn configure_drive_mode(
+    io: &DynamixelSerialIO,
+    serial_port: &mut dyn SerialPort,
+    puppet: bool,
+) -> Vec<bool> {
+    println!("------Configuring drive mode");
+
+    // get current positions
+    let pos = get_positions(io, serial_port, puppet);
+
+    // get nearest rounded positions
+    let nearest_rounded = get_nearest_rounded_positions(&pos);
+
+    // if a position of a servo is not exactly the one wanted for position2, we need to invert the mode of this servo by setting the drive mode to 1
+    let inverted = nearest_rounded
+        .iter()
+        .zip(wanted_pos2().iter())
+        .map(|(x, y)| x != y)
+        .collect::<Vec<_>>();
+
+    set_drive_mode(io, serial_port, puppet, &inverted);
+
+    return inverted;
+}
+
+fn main() {
+    let cli = Cli::parse();
+
+    let mut serial_port = serialport::new(cli.port, 1_000_000)
+        .timeout(Duration::from_secs(5))
+        .open()
+        .expect("Failed to open port");
```

Comment:
A catchable error with maybe explaination of how to debug like use Dynamixel wizard could be nicer.